### PR TITLE
Development

### PR DESCRIPTION
**Pull request by musaeed**
Full authorization feature supporting Oauth2.0 protocol + cleaned up some un-polished and un-needed code.

### CODE REVIEW

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.InMemoryTokenStore;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+import javax.sql.DataSource;
+
+/**
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:28]: please remove the names comments on top. Its just redundant. With modern tools, you know who wrote the code.

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        private CustomAuthenticationEntryPoint customAuthenticationEntryPoint;
+
+        @Autowired
+        private CustomLogoutSuccessHandler customLogoutSuccessHandler;
+
+        @Override
+        public void configure(HttpSecurity http) throws Exception {
+
+
+            http
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:48]: identations!

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    }
+
+    @Configuration
+    @EnableAuthorizationServer
+    protected static class Authorization extends AuthorizationServerConfigurerAdapter implements EnvironmentAware{
+
+
+        private static final String ENV_OAUTH = "authentication.oauth.";
+        private static final String PROP_CLIENTID = "clientid";
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:76]: are these default values.
Avoid hardcoding.

File: src/main/java/com/jersey/Authorization/Config/SecurityConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    @Bean
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @EnableGlobalMethodSecurity(prePostEnabled = true, jsr250Enabled = true)
+    private static class GlobalSecurityConfiguration extends GlobalMethodSecurityConfiguration {
+
+        public GlobalSecurityConfiguration(){
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/Config/SecurityConfiguration.java:68]: dont need an empty constructor, if no other constructor has been defined.

File: src/main/java/com/jersey/Authorization/security/AccountUserDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Created by muhammad on 6/11/17.
+ */
+@Component("userDetailsService")
+public class AccountUserDetailsService implements UserDetailsService {
+
+
+    @Autowired
+    private LoginDao accountRepository;
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/security/AccountUserDetailsService.java:31]: non-intuitive name. I would never guess that accountRepository is a loginDao instance.

File: src/main/java/com/jersey/Authorization/security/Authorities.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.jersey.Authorization.security;
+
+/**
+ * Created by muhammad on 6/13/17.
+ */
+public enum Authorities {
+
+
+    ROLE_ANONYMOUS,
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/security/Authorities.java:9]: we dont need anonymous or admin. We have only one user. Nobody has admin rights.

File: src/main/java/com/jersey/Authorization/security/Authority.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@Entity
+public class Authority {
+
+    @Id
+    @NotNull
+    @Size(min = 0, max = 50)
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/security/Authority.java:16]: should be controlled in database end. Lets keep a standard. 
Either we do it here or we do it at database end.

File: src/main/java/com/jersey/Authorization/security/CustomLogoutSuccessHandler.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    public void onLogoutSuccess(HttpServletRequest request,
+                                HttpServletResponse response,
+                                Authentication authentication)
+            throws IOException, ServletException {
+
+        String token = request.getHeader(HEADER_AUTHORIZATION);
+
+        if (token != null && token.startsWith(BEARER_AUTHENTICATION)) {
+
+            OAuth2AccessToken oAuth2AccessToken = tokenStore.readAccessToken(token.split(" ")[0]);
```
Comment (yar-malik) [inline src/main/java/com/jersey/Authorization/security/CustomLogoutSuccessHandler.java:42]: Can a token be generated with multiple spaces?

File: src/main/java/com/jersey/config/SqlInitialization.java [hunk 9]
```diff
@@ -30,6 +32,7 @@
     public DataSource dataSource() {
 
         URI dbUri = null;
+        String PALSPLATE_DB_URL = "postgres://vrwuksarnhwksl:94a6d7413fcd44094beab4b6b2d4ef0ee2dfe4b5caecf9a3d35e17b6bf6eab5b@ec2-79-125-2-69.eu-west-1.compute.amazonaws.com:5432/dbnane6c19acb5";
```
Comment (yar-malik) [inline src/main/java/com/jersey/config/SqlInitialization.java:35]: We both did this. Lets remove it, one day.

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,8 @@
+#web server
+server.port=8080
+server.contextPath=/api
+
+#Application specific
+authentication.oauth.clientid=webclient
+authentication.oauth.secret=secret
+authentication.oauth.tokenValidityInSeconds=1800
```
Comment (yar-malik) [inline src/main/resources/application.properties:8]: 30 min? Its too much. Lets make it 5 min

File: src/main/java/com/jersey/persistence/PersonDao.java [hunk 11]
```diff
@@ -2,7 +2,11 @@
 
 import com.jersey.representations.Person;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 
 public interface PersonDao extends JpaRepository<Person, Long> {
+
```
Comment (yar-malik) [inline src/main/java/com/jersey/persistence/PersonDao.java:9]: ?

File: src/main/java/com/jersey/config/SqlInitialization.java [hunk 12]
```diff
@@ -42,15 +45,21 @@ public DataSource dataSource() {
         log.info(password);
 
         String dbUrl = "jdbc:postgresql://" + dbUri.getHost() + ':' + dbUri.getPort() + dbUri.getPath() + "?sslmode=require";
```
Comment (yar-malik) [inline src/main/java/com/jersey/config/SqlInitialization.java:47]: redundant. please remove. Play around with PALSPLATE_DB_URL

File: src/main/java/com/jersey/persistence/LoginDao.java [hunk 13]
```diff
@@ -1,7 +1,15 @@
 package com.jersey.persistence;
 
 import com.jersey.representations.Login;
+import com.jersey.representations.Person;
```
Comment (yar-malik) [inline src/main/java/com/jersey/persistence/LoginDao.java:4]: remove

File: src/main/java/com/jersey/persistence/LoginDao.java [hunk 14]
```diff
@@ -1,7 +1,15 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.Optional;
```
Comment (yar-malik) [inline src/main/java/com/jersey/persistence/LoginDao.java:9]: remove

File: src/main/java/com/jersey/persistence/PersonDao.java [hunk 15]
```diff
@@ -2,7 +2,11 @@
 
 import com.jersey.representations.Person;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
```
Comment (yar-malik) [inline src/main/java/com/jersey/persistence/PersonDao.java:5]: remove

File: src/main/java/com/jersey/persistence/LoginDao.java [hunk 16]
```diff
@@ -1,7 +1,15 @@
...
 
 public interface LoginDao extends JpaRepository<Login, Long> {
+
+    @Query("SELECT u FROM Login u WHERE LOWER(u.username) = LOWER(:username)")
```
Comment (yar-malik) [inline src/main/java/com/jersey/persistence/LoginDao.java:13]: why do you need this? 
Cant you create a method in LoginResource. What is the benefit of this @Query.
Do educate me on this matter.

**Review by yar-malik — CHANGES_REQUESTED** [review]

**Comment by yar-malik** [comment]
Closing the pull request

File: src/main/java/com/jersey/persistence/LoginDao.java (hunk 16 above)
Comment (yar-malik) [reply src/main/java/com/jersey/persistence/LoginDao.java:13]: i think you are most probably right here.

Comment (yar-malik) [reply src/main/java/com/jersey/persistence/LoginDao.java:13]: Your approach is right. I looked at some cool project codes and thats how they do it. Cheers. Query can do really complicated stuff.

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java (hunk 1 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:28]: I didn't add them, they were automatically added by the ide :D

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java (hunk 2 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:48]: fixed

File: src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java (hunk 3 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/Config/OAuth2Configuration.java:76]: didn't find any other way until now

File: src/main/java/com/jersey/Authorization/Config/SecurityConfiguration.java (hunk 4 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/Config/SecurityConfiguration.java:68]: Doesn't compile without it

File: src/main/java/com/jersey/Authorization/security/AccountUserDetailsService.java (hunk 5 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/security/AccountUserDetailsService.java:31]: fixed

File: src/main/java/com/jersey/Authorization/security/Authorities.java (hunk 6 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/security/Authorities.java:9]: removed

File: src/main/java/com/jersey/Authorization/security/Authority.java (hunk 7 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/security/Authority.java:16]: removed

File: src/main/java/com/jersey/Authorization/security/CustomLogoutSuccessHandler.java (hunk 8 above)
Comment (musaeed) [reply src/main/java/com/jersey/Authorization/security/CustomLogoutSuccessHandler.java:42]: nope

File: src/main/java/com/jersey/config/SqlInitialization.java (hunk 9 above)
Comment (musaeed) [reply src/main/java/com/jersey/config/SqlInitialization.java:35]: haha yeah

File: src/main/java/com/jersey/config/SqlInitialization.java (hunk 12 above)
Comment (musaeed) [reply src/main/java/com/jersey/config/SqlInitialization.java:47]: this is not from me, maybe you can do it in your next change?

File: src/main/resources/application.properties (hunk 10 above)
Comment (musaeed) [reply src/main/resources/application.properties:8]: This means if a user stays on the site for more than 5 min, they will be logged out. 30 min is very reasonable.

File: src/main/java/com/jersey/persistence/PersonDao.java (hunk 15 above)
Comment (musaeed) [reply src/main/java/com/jersey/persistence/PersonDao.java:5]: done

File: src/main/java/com/jersey/persistence/LoginDao.java (hunk 13 above)
Comment (musaeed) [reply src/main/java/com/jersey/persistence/LoginDao.java:4]: removed

File: src/main/java/com/jersey/persistence/LoginDao.java (hunk 14 above)
Comment (musaeed) [reply src/main/java/com/jersey/persistence/LoginDao.java:9]: done

File: src/main/java/com/jersey/persistence/LoginDao.java (hunk 16 above)
Comment (musaeed) [reply src/main/java/com/jersey/persistence/LoginDao.java:13]: yup

File: src/main/resources/application.properties (hunk 10 above)
Comment (yar-malik) [reply src/main/resources/application.properties:8]: okay yea. We can always easily change it later.

**Pull request closed without merging**